presto-hive/src/main/java/com/facebook/presto/hive/PrestoS3FileSystem.java

```diff
@@ -326,7 +326,12 @@ public FSDataOutputStream create(Path path, FsPermission permission, boolean ove
             throw new IOException("File already exists:" + path);
         }
 
-        createDirectories(stagingDirectory.toPath());
+        if (!stagingDirectory.exists()) {
+            createDirectories(stagingDirectory.toPath());
+        }
+        if (!stagingDirectory.isDirectory()) {
+            throw new IOException("Staging directory is not a directory: " + stagingDirectory);
```

Comment:
Isn't the message `Staging directory is not a directory` confusing for the average user? How about something like "staging path doesn't point to a valid directory" or sth along those lines?